dashboard/src/pages/hardwareDetails/HardwareDetailsHeaderTable.tsx

```diff
@@ -191,9 +191,7 @@ export function HardwareHeader({
     'hardwareDetailsTrees',
   );
 
-  const [rowSelection, setRowSelection] = useState(() =>
-    getInitialRowSelection(selectedIndexes, treeItems.length),
-  );
+  const [rowSelection, setRowSelection] = useState({});
```

Comment:
since useState is not receiving `rowSelection` in the dependency array, is not validating when the state is different from here, 
I believe you should keep
```
const [rowSelection, setRowSelection] = useState(() =>
    getInitialRowSelection(selectedIndexes, treeItems.length),
  );
```
as initial state